### PR TITLE
Moving Blacklist policies to a new page

### PR DESCRIPTION
**Pull request by kavishkafernando**
This PR includes the following changes

- Moving Blacklist policies to a new page
- Put the "The following sample siddhi query ..." text into an infobox
- Add key template helper text
- Validate Key Templates from UI

**Define policy:**
Screen Shot 2020-06-01 at 2 33 58 PM

**Edit policy:**
Screen Shot 2020-06-01 at 2 34 24 PM

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx [hunk 1]
```diff
@@ -100,8 +120,22 @@ function AddEdit(props) {
     const [validationError, setValidationError] = useState([]);
     const [editMode, setIsEditMode] = useState(false);
     const restApi = new API();
+    const { policyId } = props.match.params;
 
     useEffect(() => {
+        if (policyId !== 'create') {
+            setIsEditMode(true);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx:127]: I think we can set this state at the state initialization time 
i:e
```javascript
const [editMode, setIsEditMode] = useState(policyId !== 'create');
```

Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx]: Fixed with `6317b1b`

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx [hunk 2]
```diff
@@ -100,8 +120,22 @@ function AddEdit(props) {
...
+            restApi.customPolicyGet(policyId).then((result) => {
+                const formattedSiddhiQuery = sqlFormatter.format(result.body.siddhiQuery);
+                const editState = {
+                    policyName: result.body.policyName,
+                    description: result.body.description,
+                    keyTemplate: result.body.keyTemplate,
+                    siddhiQuery: formattedSiddhiQuery,
+                };
+                dispatch(editState);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx:135]:
```javascript
                dispatch(editState);
```
Though this `dispatch` call does the job here, It's better to follow the action dispatching methodology which is described in [Flux pattern](https://facebook.github.io/flux/docs/in-depth-overview#actions).
In the Flux pattern Object that we pass into the `dispatcher` called an action, Typically there are two elements in an action type and payload, and we derive the next state by switching the executing by the action type and incoming event's payload.

This pattern can be seen in [React documentation examples](https://reactjs.org/docs/hooks-reference.html#lazy-initialization) as well.

Shall we follow a similar pattern here as well?

Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx]: Fixed with `4e14418`

**Pull request merged**